src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -77,14 +74,154 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
+    }
+  }
+
+  @Override
+  synchronized public void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+    log.info(String.format("Received %d offers", offers.size()));
+
+    if (liveState.getStagingTasksSize() != 0) {
+      log.info("Declining offers because tasks are currently staging");
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+    } else {
+      boolean acceptedOffer = false;
+      for (Offer offer : offers) {
+        if (acceptedOffer) {
+          driver.declineOffer(offer.getId());
+        } else {
+          switch (liveState.getCurrentAcquisitionPhase()) {
+            case JOURNAL_NODES :
+              if (liveState.getJournalNodeSize() < conf.getJournalNodeCount()) {
+                if (maybeLaunchJournalNode(driver, offer)) {
+                  acceptedOffer = true;
+                } else {
+                  driver.declineOffer(offer.getId());
+                }
+              }
+              break;
+            case NAME_NODE_1 :
+              if (maybeLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case NAME_NODE_2 :
+              if (maybeLaunchNameNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+            case DATA_NODES :
+              if (maybeLaunchDataNode(driver, offer)) {
+                acceptedOffer = true;
+              } else {
+                driver.declineOffer(offer.getId());
+              }
+              break;
+          }
+        }
+      }
+    }
+  }
+
+  @Override
+  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
+    log.info("Slave lost slaveId=" + slaveId.getValue());
+  }
+
+  @Override
+  public void run() {
+    FrameworkInfo.Builder frameworkInfo = FrameworkInfo.newBuilder()
+        .setName("HDFS " + conf.getClusterName())
+        .setFailoverTimeout(conf.getFailoverTimeout())
+        .setUser(conf.getHdfsUser())
+        .setRole(conf.getHdfsRole())
+        .setCheckpoint(true);
+
+    try {
+      FrameworkID frameworkID = persistentState.getFrameworkID();
+      if (frameworkID != null) {
+        frameworkInfo.setId(frameworkID);
+      }
+    } catch (InterruptedException | ExecutionException | InvalidProtocolBufferException e) {
+      throw new RuntimeException(e);
+    }
+
+    MesosSchedulerDriver driver = new MesosSchedulerDriver(this, frameworkInfo.build(),
+        conf.getMesosMasterUri());
+    driver.run().getValueDescriptor().getFullName();
+  }
   private void launchNode(SchedulerDriver driver, Offer offer,
-      String nodeName, List<String> taskNames, String executorName) {
+                          String nodeName, List<String> taskNames, String executorName) {
```

Comment:
This is how it's indented elsewhere. Ex: https://github.com/mesosphere/hdfs/blob/master/src/main/java/org/apache/mesos/hdfs/Scheduler.java#L53

Which one do we want. I prefer this style of aligning with the (